src/prefect/serialization/result.py

```diff
@@ -152,9 +116,13 @@ class StateResultSchema(OneOfSchema):
         "PrefectResult": PrefectResultSchema,
         "S3Result": S3ResultSchema,
         "SecretResult": SecretResultSchema,
-        "ResultHandlerResult": ResultHandlerResultSchema,
     }
 
+    def _load(self, data, *args, **kwargs):
+        if data.get("type") not in self.type_schemas:
+            data["type"] = "CustomResult"
```

Comment:
these are the two key lines for ensuring we can deserialize old-style results / result handlers